### PR TITLE
SE-650: Switch xlrd to openpyxl

### PR DESCRIPTION
**Pull request by mneedham**
# Pull Request Checklist

- [x] Read the [contributing guidelines](https://github.com/finbourne/lusid-python-tools/blob/master/docs/CONTRIBUTING.md)
- [x] Tests pass

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

**Comment by rizwansaeed** [comment]
does this mean all current usages need to specify `engine="openpyxl"` or is the default still `xlrd` and specifying the engine allows you to take advantage of the update?

**Comment by mneedham** [comment]
Yeh the default does still seem to be `xlrd`:

```
    def __init__(self, path_or_buffer, engine=None):
        if engine is None:
            engine = "xlrd"
            if isinstance(path_or_buffer, (BufferedIOBase, RawIOBase)):
                if _is_ods_stream(path_or_buffer):
                    engine = "odf"
            else:
                ext = os.path.splitext(str(path_or_buffer))[-1]
                if ext == ".ods":
                    engine = "odf"
        if engine not in self._engines:
            raise ValueError(f"Unknown engine: {engine}")
```

Although presumably that would then throw an exception if the `xlrd` library isn't installed...

File: requirements.txt [hunk 1]
```diff
@@ -9,4 +9,4 @@ pandas==1.1.4
 PyYAML==5.4
 requests==2.25.0
 tqdm==4.52.0
-xlrd==1.2.0
```
Comment (thomasdotred) [inline requirements.txt:12]: Can we leave this dependency in just in case some of the other tests (for cocoon) that read excel files?

**Review by thomasdotred — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (mneedham) [reply requirements.txt:12]: Sure.

**Review by thomasdotred — APPROVED** [review]

**Comment by thomasdotred** [comment]
waiting for build to pass

**Pull request merged**